test/Libraries/Microsoft.AspNetCore.Telemetry.Middleware.Tests/Metering/HttpMeteringTests.cs

```diff
@@ -49,7 +49,7 @@ public HttpMeteringTests()
     [Fact]
     public async Task CanLogIncomingRequestMetric()
     {
-        using var metricCollector = new MetricCollector(new List<string> { typeof(HttpMeteringMiddleware).FullName! });
+        using var metricCollector = new MetricCollector<long>(null, typeof(HttpMeteringMiddleware).FullName!, Metric.IncomingRequestMetricName);
```

Comment:
I lost track of that other comment that was talking about using scopes but the basic pattern is this:
1. When production code that is DI-aware creates a Meter, it should do so by taking an IMeterFactory as an injected constructor parameter and then creating the Meter from the factory. That factory now serves as the scope for the Meter.
2. When you want to test later, get the factory out of the DI container and use it as the scope parameter to the collector. This ensures that the Meter the test binds to is the one created by the factory in your test's container and not one that was created in a different DI container from a parallel test.

(If you have a way to get a reference to the Meter or a reference to the Instrument that works fine too, but often the components which create Meters and instruments don't expose those references)